tests/README.md

```diff
@@ -0,0 +1,165 @@
+
+**Auth Fixtures**
+
+Since the NASA APT frontend uses multiple users, we have created a Page object that allows us to login as different users each with a different role. This allows us to store the login credentials in localstorage and load them back up when we need to login as the user with that particular role. The authentication mechanism is the following:
+
+1. We generate the `storageState` of the user by logging in as the user and storing the localstorage of the user in a JSON file. This is done in the initial tests found in `tests/auth.setup.js`. The `storageState` is stored in `playwright/.auth/` folder.
+
+2. We extend the `Page` class from Playwright to create a `LoggedInPage` class that has a `login` method that allows us to login as a user with a particular role. The `login` method takes in a `role` parameter that is used to determine which user to login as. The `LoggedInPage` takes a context which is initialized with the `storageState` of the user. The `storageState` is a JSON file that contains the localstorage of the user. This allows us to login as the user without having to enter the credentials each time.
+
+```js
+  ownerPage: async ({ browser }, use) => {
+    const context = await browser.newContext({
+      storageState: 'playwright/.auth/owner.json'
+    });
+    const ownerPage = new LoggedInPage(await context.newPage(), 'owner');
+    await ownerPage.init();
+    await use(ownerPage);
+  },
+```
+
+The init function calls `.login()` on the page object which logs in the user with the role specified in the constructor. It also calls `.routes()` which sets up the routes for the page object. The routes are used to intercept requests and responses and perform actions based on the request and response.
+
+3. In a test where we need to use the owner credentials, instead of using the standard `test` function from Playwright, we use the `ownerPage` fixture that we created. This allows us to login as the owner user and use the `ownerPage` object to interact with the app.
+
+```js
+import { test, expect } from '../../playwright/fixtures';
+
+test('should be able to login as owner', async ({ ownerPage }) => {
+  const { page } = ownerPage;
+  await page.goto('/dashboard');
+  await expect(page).toHaveText('h1', 'Dashboard');
+});
+```
+
+### Writing tests
+
+**Playwright tools for writing tests**
+
+Playwright differs from other testing frameworks by using accessibility selectors to locate elements on the page. This allows us to write tests that are more resilient to changes in the DOM. When you write a test, you can use the record mechanism to record the behavior of the user on the page as well as select elements on the page to use as locators. You can learn more about VSCode here: [https://playwright.dev/docs/getting-started-vscode](https://playwright.dev/docs/getting-started-vscode)
```

Comment:
Maybe, the section about VS code be separated, it looks a bit out of place at the moment.